src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -777,8 +777,9 @@
       if (excludeBaseClasses && class %in% c("list", "environment"))
          next
       
-      methodName <- paste(".DollarNames", class, sep = ".")
-      method <- .rs.getAnywhere(methodName, envir = envir)
+      method <- utils::getS3method(".DollarNames", class, optional = TRUE, 
+        envir = getNamespace("utils"))
```

Comment:
Right, but the environment is where the method searches for the underlying method, not generic, e.g. from `getS3method` definition:

```
method <- paste(f, class, sep = ".")
if (!is.null(m <- get0(method, envir = envir, mode = "function")))
    return(m)
```